### ISSUE TITLE
Namespaces.UnusedUses.UnusedUse removes despite being used in annotations in code, e.g. @var

### ISSUE THREAD

**Issue by kkmuffme** [slevomat/coding-standard#1530]
The rule removes "use" statements, despite them being used in annotations inside the code, e.g. @var

Config:

```
<rule ref="SlevomatCodingStandard.Namespaces.UnusedUses">
	<properties>
		<property name="searchAnnotations" value="true"/>
	</properties>
</rule>
```

e.g. common when type guarding external libs/using static analysis:

```php
use Foo\Bar;

/**
 * @var Bar
 */
$foo = some_func();
```

or also a very common case in e.g. WordPress:

```php
use Foo\Bar;

/**
 * @param Bar|false
 */
$foo = apply_filters( 'hello', false );
```

or

```php
use Foo\Bar;

/**
 * @param Bar|false $arg
 */
$foo = apply_filters( 'hello', $arg );
```

or

```php
use Foo\Bar;

/**
 * @param int|false
 * @param Bar $arg
 */
$foo = apply_filters( 'hello', false, $arg );
```

**Comment by kukulich** [comment]
Please provide a PR with failing test.

**Comment by kukulich** [comment]
The bug is caused by https://github.com/phpstan/phpdoc-parser - there's no support for `@param` without parameter name.

**Comment by kukulich** [comment]
Yes, it's probably reported as `InvalidTagValueNode`. However I don't want to add support for invalid tags. I use https://github.com/phpstan/phpdoc-parser so I don't have to do annotations parsing by myself.

**Comment by TakeruDavis** [comment]
I would like to add to this a very similar problem with this rule, it doesn't work with generics in annotations either and removes the import

```
use Foo\Bar;

/**
 * @var \Collection<Bar>
 */
$foo = some_func();
```

**Comment by kukulich** [comment]
@TakeruDavis Generics should work. Please try to prepare PR with failing test.

**Comment by TakeruDavis** [comment]
@kukulich False alarm, looks like we've been locked on an older version of the package
